materialize-sql/type_mapping.go

```diff
@@ -169,6 +231,47 @@ func WithElementConverter(converter ElementConverter) StaticMapperOption {
 	}
 }
 
+// StringCastConverter builds an ElementConverter from the string-handling callback. Any non-string
+// types are returned without modification. This should be used for converting fields that have a
+// string type and one additional type. The callback should convert the string into the desired type
+// per the endpoint's requirements.
+func StringCastConverter(fn func(string) (interface{}, error)) ElementConverter {
```

Comment:
Technically everything seems to work fine with all our current SQL materializations if we don't do this, and just let the database handle an `interface{}` with a concrete type of `string` as its input for a numeric column. Insert queries into postgres (for example) when doing this would sometimes use `1` and sometimes use `'1'` as their parameter value into a `BIGINT` column. This does apparently work but feels really wrong, and may cause bugs I haven't tested for in our current SQL materializations, or perhaps future ones. So I am making it require a little more work to get the correct type for the queries.